server/src/sas/CompletionProvider.ts

```diff
@@ -281,6 +284,381 @@ export class CompletionProvider {
+                index = match.index;
+                _match = match;
+              }
+              // the last match
+              if (_match) {
+                syntaxArr.push(data.syntax.slice(_match.index));
+              }
+              // no function overloading, push the first match
+              if (!syntaxArr.length) {
+                syntaxArr.push(data.syntax);
+              }
+
+              // match arguments
+              // clean syntax
+              syntaxArr = syntaxArr.map((syntax) =>
+                syntax
+                  .replace(/\*/g, "")
+                  .replace(/<sub>/g, "")
+                  .replace(/<\/sub>/g, "")
+                  .replace(/&lt;/g, "<")
+                  .replace(/&gt;/g, ">")
+                  .replace(/\s+/g, " ")
+                  .replace(/\s*Form \d:\s*/gi, ""),
+              );
+              // arguments in syntax
+              // allArguments is used for filter arguments in function overloading
+              const allArguments: string[] = [];
+              const syntaxArgumentArr = syntaxArr.map(
+                (syntax) =>
+                  syntax
+                    .match(/\((.*)\)/)?.[1]
+                    ?.split(",")
+                    .map((argument) => {
+                      let _arguments = [argument];
+                      // match like time|datetime, interval <multiple><.shift-index> or <<btrim-specification><'btrim-character' FROM>> sql-expression
+                      if (
+                        argument.includes("|") ||
+                        (argument.includes(">") &&
+                          argument.includes("<") &&
+                          (argument.indexOf(">") > 0 ||
+                            argument.indexOf("<") > 0))
+                      ) {
+                        let match: any;
+                        let _match: any;
+                        let _index = 0;
+                        const _argument = argument;
+                        const regexp = RegExp("[<>\\|]", "g");
+                        _arguments = [];
+                        while ((match = regexp.exec(_argument)) !== null) {
+                          _arguments.push(argument.slice(_index, match.index));
+                          _index = match.index;
+                          _match = match;
+                        }
+                        // the last match
+                        if (_match) {
+                          _arguments.push(argument.slice(_match.index));
+                        }
+                      }
+                      _arguments = _arguments.map(
+                        (_argument) =>
+                          _argument
+                            ?.replace("(s)", "")
+                            .replace(/['"‘’<>()|.]/g, "")
+                            .replace(/[-_\s]?\d$/, "")
+                            .replace(/[-_\s]n$/, "")
+                            .trim()
+                            .toLowerCase(),
```

Comment:
looks like used at several places, consider extract to a function like `cleanupArg`